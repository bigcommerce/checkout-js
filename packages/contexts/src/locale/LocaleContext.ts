import type { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);
