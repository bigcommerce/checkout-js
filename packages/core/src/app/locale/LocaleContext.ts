import { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

const LocaleContext = createContext<LocaleContextType | undefined>(undefined);

export default LocaleContext;
