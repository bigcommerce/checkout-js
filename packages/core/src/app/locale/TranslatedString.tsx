import React, { Fragment, FunctionComponent } from 'react';

import withLanguage, { WithLanguageProps } from './withLanguage';

export interface TranslatedStringProps {
    id: string;
    data?: any;
}

const TranslatedString: FunctionComponent<TranslatedStringProps & WithLanguageProps> = ({
    data,
    id,
    language,
}) => (
    <Fragment>
        { language.translate(id, data) }
    </Fragment>
);

export default withLanguage(TranslatedString);
