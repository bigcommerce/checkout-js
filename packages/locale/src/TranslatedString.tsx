import React, { FunctionComponent } from 'react';

import withLanguage, { WithLanguageProps } from './withLanguage';

export interface TranslatedStringProps {
    id: string;
    data?: any;
}

const TranslatedString: FunctionComponent<TranslatedStringProps & WithLanguageProps> = ({
    data,
    id,
    language,
}) => <>{language.translate(id, data)}</>;

export default withLanguage(TranslatedString);
