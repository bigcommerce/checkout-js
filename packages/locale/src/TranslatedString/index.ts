export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
