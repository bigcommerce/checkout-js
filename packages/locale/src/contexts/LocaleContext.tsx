import { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext, useContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);

export function useLocale() {
    const context = useContext(LocaleContext);

    if (!context) {
        throw new Error('useLocale must be used within a LocaleContextProvider');
    }

    return context;
}
