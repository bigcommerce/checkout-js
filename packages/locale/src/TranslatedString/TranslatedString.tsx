import { TranslationData } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent } from 'react';

import { useLocale } from '../contexts';

export interface TranslatedStringProps {
    id: string;
    data?: TranslationData;
}

const TranslatedString: FunctionComponent<TranslatedStringProps> = ({ data, id }) => {
    const { language } = useLocale();

    return <>{language.translate(id, data)}</>;
};

export default TranslatedString;
