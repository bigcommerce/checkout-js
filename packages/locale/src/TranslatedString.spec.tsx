import React from 'react';
import testRenderer from 'react-test-renderer';

import LocaleContext from './LocaleContext';
import { getLocaleContext } from './localeContext.mock';
import TranslatedString from './TranslatedString';

describe('TranslatedString Component', () => {
    const localeContext = getLocaleContext();

    jest.spyOn(localeContext.language, 'translate');

    it('renders translated string', () => {
        expect(
            testRenderer
                .create(
                    <LocaleContext.Provider value={localeContext}>
                        <TranslatedString id="address.address_line_1_label" />
                    </LocaleContext.Provider>,
                )
                .toJSON(),
        ).toMatchSnapshot();
    });

    it('calls language.translate', () => {
        expect(
            testRenderer
                .create(
                    <LocaleContext.Provider value={localeContext}>
                        <TranslatedString id="address.address_line_1_label" />
                    </LocaleContext.Provider>,
                )
                .toJSON(),
        );

        expect(localeContext.language.translate).toHaveBeenCalledWith(
            'address.address_line_1_label',
            undefined,
        );
    });

    it('calls language.translate with data when passed', () => {
        const data = { foo: 'xyz' };

        expect(
            testRenderer
                .create(
                    <LocaleContext.Provider value={localeContext}>
                        <TranslatedString data={data} id="address.address_line_1_label" />
                    </LocaleContext.Provider>,
                )
                .toJSON(),
        );

        expect(localeContext.language.translate).toHaveBeenCalledWith(
            'address.address_line_1_label',
            data,
        );
    });
});
