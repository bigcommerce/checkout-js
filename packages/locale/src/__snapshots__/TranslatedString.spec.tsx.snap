// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslatedString Component renders translated string 1`] = `"Address"`;
